kernel/aster-nix/src/process/process_vm/mod.rs

```diff
@@ -14,30 +14,42 @@ use user_heap::UserHeap;
 use crate::vm::vmar::Vmar;
 
 /*
-* The user vm space layout is look like below.
-* |-----------------------|-------The highest user vm address
-* |                       |
-* |       Mmap Areas      |
-* |                       |
-* |                       |
-* --------------------------------The init stack base
-* |                       |
-* | User Stack(Init Stack)|
-* |                       |
-* |         ||            |
-* ----------||----------------------The user stack top, grows down
-* |         \/            |
-* |                       |
-* |     Unmapped Areas    |
-* |                       |
-* |         /\            |
-* ----------||---------------------The user heap top, grows up
-* |         ||            |
-* |                       |
-* |        User Heap      |
-* |                       |
-* ----------------------------------The user heap base
-*/
+ * The user's virtual memory space layout looks like below.
+ * TODO: The layout of the userheap does not match the current implementation,
+ * And currently the initial program break is a fixed value.
+ *
+ *  (high address)
+ *  +---------------------+ <------+ The top of Vmar, which is the highest address usable
+ *  | A reserved page     |          // FIXME: This should not be mappable but we neglected
```

Comment:
This is a very good point. I never knew that the last page in the lower half of virtual memory was unmappable. In fact, it's a workaround for some nasty CPU bugs, both Intel's and AMD's, as explained in the [Linux kernel source](https://github.com/torvalds/linux/blob/480e035fc4c714fb5536e64ab9db04fedc89e910/arch/x86/include/asm/page_64.h#L68-L78).

But if this is the case, why not just set `USERSPACE_LOWEST_UNUSABLE_VADDR` to `0x0000_7fff_ffff_f000`? This is exactly what Linux did for its [`task_size_max()` macro](https://github.com/torvalds/linux/blob/480e035fc4c714fb5536e64ab9db04fedc89e910/arch/x86/include/asm/page_64_types.h#L61). 

As far as I know, this is an architecture-specific workaround, and the guard page doesn't exist for other architectures. So encoding it in the `USERSPACE_LOWEST_UNUSABLE_VADDR` constant seems a bit more correct. Meanwhile, perhaps the above explanation from Linux kernel needs to be referenced to make things clearer.